diffyscan/utils/binary_verifier.py

```diff
@@ -0,0 +1,142 @@
+
+    for ind in list(checkpoints):
+        startIndex = max(0, ind - nearLinesCount)
+        lastIndex = min(ind + nearLinesCount, maxInstructionsCount - 1)
+        for i in range(startIndex, lastIndex + 1):
+            checkpoints.add(i)
+
+    checkpointsArray = sorted(list(checkpoints))
+
+    logger.divider()
+    logger.info(f"0000 00 STOP - both expected and actual bytecode instructions match")
+    logger.info(f'{bgRed("0x0002")} - the actual bytecode differs')
+    logger.info(
+        f'{bgYellow("0x0001")} - the actual bytecode differs on the immutable reference position'
+    )
+    logger.info(
+        f'{bgGreen("0x0003")} - the expected bytecode value when it doesn\'t match the actual one'
+    )
+    logger.info(
+        f'{red("0000 00 STOP")} - the actual bytecode instruction doesn\'t exist, but expected is present'
+    )
+    logger.info(
+        f'{green("0000 00 STOP")} - the actual bytecode instruction exists when the expected doesn\'t'
+    )
+    logger.divider()
+    for i in range(len(checkpointsArray)):
+        currInd = checkpointsArray[i]
+        prevInd = checkpointsArray[i - 1] if i > 0 else None
+        if prevInd and prevInd != currInd - 1:
+            print("...")
+
+        actual = (
+            actualInstructions[currInd] if currInd < len(actualInstructions) else None
+        )
+        expected = (
+            expectedInstructions[currInd]
+            if currInd < len(expectedInstructions)
+            else None
+        )
+
+        if not actual and expected:
+            params = "0x" + expected["bytecode"][2:]
+            print(
+                logger.red(
+                    f'{to_hex(currInd, 4)} {to_hex(expected["op"]["code"])} {expected["op"]["name"]} {params}'
+                )
+            )
+        elif actual and not expected:
+            params = "0x" + actual["bytecode"][2:]
+            print(
+                logger.green(
+                    f'{to_hex(currInd, 4)} {to_hex(actual["op"]["code"])} {actual["op"]["name"]} {params}'
+                )
+            )
+        elif actual and expected:
+            opcode = (
+                to_hex(actual["op"]["code"])
+                if actual["op"]["code"] == expected["op"]["code"]
+                else bgRed(to_hex(actual["op"]["code"]))
+                + " "
+                + bgGreen(to_hex(expected["op"]["code"]))
+            )
+            opname = (
+                actual["op"]["name"]
+                if actual["op"]["name"] == expected["op"]["name"]
+                else bgRed(actual["op"]["name"]) + " " + bgGreen(expected["op"]["name"])
+            )
+            actualParams = (
+                "0x" + actual["bytecode"][2:] if len(actual["bytecode"]) > 2 else ""
+            )
+            expectedParams = (
+                "0x" + expected["bytecode"][2:] if len(expected["bytecode"]) > 2 else ""
+            )
+
+            paramsLength = len(expected["bytecode"]) // 2 - 1
```

Comment:
It seems here we deal with a hex string `0xAABBCCDD` and try to count the bytes inside.
shall it be done via a little helper func to be more reader-friendly?

Reply:
done